### PR TITLE
[Model Monitoring] add threshold at the group level

### PR DESCRIPTION
**Pull request by hawestra**
example output : https://ml.azure.com/data/azureml_38cd7b89-a78f-4044-bf50-663f57e21ee2_output_data_signal_metrics/1/details?wsid=/subscriptions/e0fd569c-e34a-4249-8c24-e8d723c7f054/resourceGroups/hawestra-rg/providers/Microsoft.MachineLearningServices/workspaces/hawestra-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#exploreTab
image (2)

### CODE REVIEW

File: assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py [hunk 1]
```diff
@@ -27,12 +27,21 @@ def evaluate_metrics_threshold(
...
         metrics_to_evaluate_df.threshold_value.isNotNull()
     )
 
+    is_not_nan_metrics_threshold_df = is_not_nan_metrics_threshold_df.filter(
+        is_not_nan_metrics_threshold_df.metric_value.isNotNull()
+    )
+
     is_not_nan_metrics_threshold_df = is_not_nan_metrics_threshold_df.where(
         F.col("metric_value") > F.col("threshold_value")
     )
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py:47]: Could you please also update the line 49-50? I think the logic here is trying to `show()` all the rows, hence the union. With your change it seems some records have been dropped.
Another option is to remove these 2 lines and make it something like:
```metrics_threshold_breached_df = is_not_nan_metrics_threshold_df.where(
        F.col("metric_value") > F.col("threshold_value")
    )
    
if metrics_threshold_breached_df .count() > 0:
    ```

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py [hunk 2]
```diff
@@ -109,14 +129,30 @@ def run():
...
-            aggregated_metrics_df = aggregated_metrics_df.filter(col("metric_name")
+            aggregated_metrics_df = aggregated_metrics_df.filter(col(METRIC_NAME_COLUMN)
                                                                  != full_per_instance_score_metric_name)
+        else:
+            threshold = _get_per_instance_threshold(histogram_df, metric_name)
+            threshold_row = spark.createDataFrame(
+                    [
+                        (
+                            "",
+                            "",
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py:147]: Sorry, I am not familiar with the per instance score metrics, but I see both "group name" and "metric value" are empty here. Just curious how this works without metric values.

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py [hunk 3]
```diff
@@ -5,14 +5,20 @@
 
 import argparse
 import re
-from pyspark.sql.functions import col, udf, sum
-from pyspark.sql.types import IntegerType
+from pyspark.sql.functions import col, udf, sum, lit
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py:8]: nit: Could you please sort this line and line 9 and line 16-21?

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py (hunk 2 above)
Comment (hawestra) [reply assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py:147]: do you mean how does that work in the evaluate component ? Mathieu wrote that part and just gave me the output he wanted.

Comment (yetamsft) [reply assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_metrics/run.py:147]: Synced offline and it seems these are some metrics not for evaluation metric component's consumption, but for UI instead. Hence they are skipped in evaluation metrics.

File: assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py (hunk 1 above)
Comment (yetamsft) [reply assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py:47]: Please confirm with Mathieu whether we'd like to "show" the new metrics in the output (1st option), or don't show anything (2nd option) or keep showing the other metrics but not the new one, which IMO is kind of weird (keep it as-is).

Comment (hawestra) [reply assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py:47]: image

**Review by yetamsft — APPROVED** [review]
LGTM!

**Pull request merged**